lib/http_requests.py

```diff
@@ -160,11 +169,14 @@ def get_advisories(branch_name):
                 current_advisories.update(basis_advisories)
             current_version = basis_version
 
-        # Apply override advisories specified with 'advisories!'
-        override_advisories = yml_data[version].get('assembly', {}).get('group', {}).get('advisories!', {})
-        if override_advisories:
+        # Check for and handle 'advisories!' override
+        override_advisories = assembly_data.get('group', {}).get('advisories!', None)
+        if override_advisories is not None:
+            if not override_advisories:  # Skip versions with empty override advisories
+                continue
```

Comment:
This could be moved above, before the while loop, so that if the advisories are overridden you can skip the while loop completely.

Reply:
Hmm actually,  a inherited advisory can have overrides, so actually this is the better place. No need to move.

Reply:
Ahh the PR is making review a bit difficult. So since an inherited advisory can be overridden, this has to be moved inside the while loop.

Reply:
To clarify, the check for 'advisories!' overrides should be placed outside and after the while loop, not inside it. This placement ensures that all inherited advisories from the basis versions are processed first, and then any overrides specified in 'advisories!' are applied.

Reply:
Lets say we are looking at a particular version `4.12.5`. In doesn't have `advisories!` but it inherits from `4.12.1`, which has `advisories!` in it. i don't think the current code can handle this scenario.

Reply:
You're right. I didn't account for that..

Reply:
I will go ahead and create the ticket, will remove everything for PR but the custom type assebly filter.
Just to clarify, with your example in mind, the solution would be:
```
def get_advisories(branch_name):
    """
    Gets the list of advisories from the releases.yml file of an OpenShift release.
    :param branch_name: OpenShift branch name in ocp-build-data. eg: openshift-4.10
    :returns: List of lists containing the advisories with the z-stream version. Eg:
                            [['4.11.6', {'extras': 102175, 'image': 102174, 'metadata': 102177, 'rpm': 102173}], ... ]
    """
    url = f"{os.environ['GITHUB_RAW_CONTENT_URL']}/{branch_name}/releases.yml"
    seen = {}
    MAX_DEPTH = 3
    yml_data = get_http_data(url).get('releases', None)

    if not yml_data:
        return None

    advisory_data = []
    for version in yml_data:
        if version in seen:
            continue

        depth = 0
        current_version = version
        current_advisories, jira_link = process_version_advisories(current_version, yml_data, seen)
        if not current_advisories:
            current_advisories = {}

        while depth < MAX_DEPTH:
            depth += 1
            current_assembly_data = yml_data[current_version].get('assembly', {})
            basis_version = current_assembly_data.get('basis', {}).get('assembly')
            if not basis_version or basis_version in seen:
                break

            basis_advisories, _ = process_version_advisories(basis_version, yml_data, seen)
            if basis_advisories:
                current_advisories.update(basis_advisories)
            current_version = basis_version

            # Apply override advisories from basis version, if any
            basis_override_advisories = yml_data[basis_version].get('assembly', {}).get('group', {}).get('advisories!', None)
            if basis_override_advisories is not None:
                if not basis_override_advisories:  # Skip versions with empty override advisories
                    current_advisories.clear()
                else:
                    current_advisories.update(basis_override_advisories)

        # Final check for current version's 'advisories!' overrides
        override_advisories = yml_data[version].get('assembly', {}).get('group', {}).get('advisories!', None)
        if override_advisories is not None:
            if not override_advisories:  # If current version's override_advisories is empty
                continue  # Skip this version as it should have no advisories
            current_advisories.update(override_advisories)

        if current_advisories:  # Append only if there are non-empty advisories
            advisory_data.append([version, current_advisories, jira_link])
        seen[version] = True

    return advisory_data
```
 The key is to update the advisory information of the current version with the overrides from its basis version, if any, during each iteration of the loop. Is that correct?